plugins/module_utils/network/sonic/config/vlans/vlans.py

```diff
@@ -168,25 +169,28 @@ def _state_replaced(self, want, have, diff):
         """
         commands = []
         requests = []
-
+        # Add default values to want so it'll be properly comparable to have and re-generate diff.
+        diff = get_diff(self.deal_with_default_entries(want, method="add"), have, TEST_KEYS)
```

Comment:
The "deal_with_default_entries" invocation here seems incorrect.

If a "replaced" play doesn't specify some option and that option is not currently configured (to a non-default value, in this case), no action should occur. If it doesn't specify some option and that option is currently configured to a non-default value, it should cause it to be logically "deleted" (set to the default value, in this case), so the second pass of the "replaced" handling (the "merge" after "delete" pass) shouldn't need to take any further action. Either way, for the diff, having the option absent from the input playbook should not result in a "forward" (want vs have) diff entry. (That should only happen if the new requested value is specified and differs from the current configuration.)